lib/model/weather/weather.dart

```diff
@@ -1,37 +1,22 @@
+import 'dart:convert';
 import 'package:flutter/material.dart';
 import 'package:flutter_training/common/result.dart';
+import 'package:flutter_training/model/weather/weather_info.dart';
+import 'package:flutter_training/model/weather/weather_request.dart';
 import 'package:yumemi_weather/yumemi_weather.dart';
 
-enum WeatherCondition {
-  sunny,
-  cloudy,
-  rainy,
-}
-
-extension _WeatherConditionExt on Iterable<WeatherCondition> {
-  WeatherCondition? byNameOrNull(String name) {
-    for (final element in this) {
-      if (element.name == name) {
-        return element;
-      }
-    }
-    return null;
-  }
-}
-
 class Weather {
   const Weather(this._weatherClient);
 
   final YumemiWeather _weatherClient;
 
-  Result<WeatherCondition?, String> fetchWeather() {
+  Result<WeatherInfo?, String> fetchWeather(WeatherRequest request) {
```

Comment:
[nits]
前回気づかなかったのですが、ここは non-null にできそうです！

```suggestion
  Result<WeatherInfo, String> fetchWeather(WeatherRequest request) {
```

Reply:
ご指摘ありがとうございます！僕も気がつかなかったです...。
こちらで修正しました。！
ccde8a8ce92cfcc1df0f6a8d9722b46f1243980f